astrocut/cube_cut.py

```diff
@@ -272,17 +272,26 @@ def _fit_cutout_wcs(self, cutout_wcs, cutout_shape):
         """
 
         # Getting matched pixel, world coordinate pairs
-        pix_inds = np.array(list(product(list(range(cutout_shape[1])), list(range(cutout_shape[0])))))
+        # We don't want more than ~100 pairs, and we want them spread evenly through the iamge
+
+        y, x = cutout_shape
+        i = 1
+        while (x/i)*(y/i) > 100:
```

Comment:
Suggest adding an in-line comment on what you are doing here.  Took me a while to figure it out (if I got it right), I think this line is determining how large the spacing should be to ensure 100 points are sampled across the (user-specified) cutout along the x and y dimension?  Does this make any implict assumption about it being a square vs. an extreme rectangle?  E.g., does this work for a requested cutout that is 1 pixel in x-dimension and 50 pixels in the y-dimension?

Reply:
The fact that the while loops over both the x- and y-dimensions with a single index counter ("i") is why I wonder if it works as expected for rectangles.  If the dimension is not a square, does this technique require looping over two different index counters to account for the spacing being different between the two dimensions?  I'd have to try it out myself with a stand-alone script for sure, but wanted to document my thought on this here.

Reply:
I believe this is working as intended and I will certainly add a comment. The goal is to create an evenly spaced mesh of no more than 100 pixel coordinates to use in the fit. In practice I do this by taking the ith value in each row/column.

This is what  happens for a 5x7 cutout with i = 2:
```
xoxoxox
ooooooo
xoxoxox
ooooooo
xoxoxox
```

Where x denotes the indexes that will be used in the fit.

Does that make sense?

Reply:
Yes it does, in fact this exact thing, including the tic-tac-toe board, would be a great thing to add as an in-line comment block.

Reply:
Will do!